### PR TITLE
Added ability to sort add-ons by last tested version

### LINKED ISSUE #18440: Add ability to quickly see and sort by minimum and last tested version of add-ons in the store
### Is your feature request related to a problem? Please describe.

The minimum and last tested version information is shown in the description for add-ons which are installed.

For someone on say NVDA 2024.4.2 who wants to look at their add-ons to see which are compatible with NVDA 2025.1 before updating, you currently need to press tab three times to "other details", then down arrow four times to Minimum version and again for last tested version.  Then press shift+tab three times to get back to the add-ons list.

Or, knowing the shortcuts, press alt+o for other details, then down arrow four times to get to the minimum version, again for last tested and then alt+a to get back to the list and down arrow to the next add-on.  That is 8 keystrokes for each add-on, and they are not ordered for minimum or last tested version.

### Describe the solution you'd like

It would be worth having at least last tested version as a column in the table view of add-ons and having the ability to sort by that column.

### PR DESCRIPTION
**Pull request by nvdaes**
- **Add ability to sort add-ons by last tested NVDA version**
- **Update changelog**

### Link to issue number:

Fixes #18440
### Summary of the issue:
Sorting add-ons by last tested version can be helpful to know if an add-on is compatible with an updated version of NVDA, available but not installed yet. In this way, users can decide if they prefer to update NVDA, or wait until a compatible version of the add-on is available.
### Description of user facing changes:
In add-on lists of the add-on store, a new column with last tested NVDA version is available, and add-ons can also be sorted by this column.

### Description of developer facing changes:

None.

### Description of development approach:
In addonStoreGui, viewModels, addonList module, a new field with its corresponding text has been added to create a new column, showing the last tested NVDA version

### Testing strategy:
Open the store, switch to various tabs, check that a new column with last tested version is available, and that add-ons can be sorted by this column.

### Known issues with pull request:
None.

### Code Review Checklist:

- [x] Documentation:
  - Change log entry
  - User Documentation
  - Developer / Technical Documentation
  - Context sensitive help for GUI changes
- [x] Testing:
  - Unit tests
  - System (end to end) tests
  - Manual testing
- [x] UX of all users considered:
  - Speech
  - Braille
  - Low Vision
  - Different web browsers
  - Localization in other languages / culture than English
- [x] API is compatible with existing add-ons.
- [x] Security precautions taken.

@coderabbitai summary

### CODE REVIEW

**Comment by nvdaes** [comment]
@seanbudd , po files contain errors, but I think that this is not related to this PR, so I'll mark this as ready for review.

**Comment by CyrilleB79** [comment]
@nvdaes (author) or @SaschaCowley / @seanbudd (reviewers), what about the visual aspect of the list? Do all the columns fit in the list?

It would also be worth having the install date column in the Installed add-ons tab, and have add-ons sorted by install date. This would be very useful for trouble shouting, especially since automatic update is available.

I also wonder, at some point, if we should not give the capability to allow the user to choose which columns are displayed, as done in the detailed view of task manager, processes tab. And to be able to sort by any column.

All these additional proposals are probably off-topic with respect to this PR, but I mention theme here in case it may impact the choices done in this PR.

**Comment by nvdaes** [comment]
Cyrille wrote:

> Do all the columns fit in the list?

I'm totally blind and I cannot ensure it. But I think so, since, when I created the publication date column in other pull request, sometimes I noted that some columns weren't read, and this was fixed by changin column size.
I took this since this was not marked as sightedIdeal.

**Comment by seanbudd** [comment]
@nvdaes - Can you also ensure the last tested and minimum API versions are always available in the add-on details panel? right now they are only visible in this installed add-ons tab.

File: source/gui/addonStoreGui/viewModels/addonList.py [hunk 1]
```diff
@@ -90,6 +91,11 @@ class AddonListField(_AddonListFieldData, Enum):
 		100,
 		frozenset({_StatusFilterKey.AVAILABLE, _StatusFilterKey.UPDATE}),
 	)
+	lastTestedVersion = (
+		# Translators: The name of the column that contains the last version of NVDA tested with this add-on.
+		pgettext("addonStore", "Last tested NVDA version"),
```
Comment (seanbudd) [inline source/gui/addonStoreGui/viewModels/addonList.py:106]: #18440 also requested adding minimum version

**Comment by seanbudd** [comment]
@CyrilleB79 - the text of the column header is going to be clipped for most resolutions, but the column is fine otherwise.

I agree that you should be able to sort by any column - visible or not, and choose what columns to display of all the available information an add-on can have. But this is probably better as a new issue

File: source/gui/addonStoreGui/viewModels/addonList.py (hunk 1 above)
Comment (seanbudd) [reply source/gui/addonStoreGui/viewModels/addonList.py:106]: can you also move these after date

File: source/gui/addonStoreGui/viewModels/addonList.py [hunk 2]
```diff
@@ -90,6 +91,11 @@ class AddonListField(_AddonListFieldData, Enum):
...
+		50,
+	)
 	publicationDate = (
 		# Translators: The name of the column that contains the publication date of the add-on.
 		pgettext("addonStore", "Date"),
```
Comment (seanbudd) [inline source/gui/addonStoreGui/viewModels/addonList.py]:
```suggestion
		pgettext("addonStore", "Publication Date"),
```

Comment (seanbudd) [reply source/gui/addonStoreGui/viewModels/addonList.py]: This might be a bit long but I found I think it would be useful to know what the date means. I think it might be confusing with all the NVDA versions, when some synths read our version numbers as dates.

File: user_docs/en/changes.md [hunk 3]
```diff
@@ -6,6 +6,8 @@
 
 ### New Features
 
+* In add-on lists of the add-on store, add-ons can be sorted by minimum and last tested NVDA version. Additionally, minimum and last tested version will be shown in the Details area for all tabs of the store (#18440, @nvdaes)
```
Comment (seanbudd) [inline user_docs/en/changes.md]: I think this should go under "changes" not new features.

```suggestion
* In the Add-on Store, add-ons can be sorted by minimum and last tested NVDA version.
Additionally, minimum and last tested version will now be also shown in the details area for an add-on in the Available Add-ons tab. (#18440, @nvdaes)
```

**Review by seanbudd — APPROVED** [review]
